http-server/src/main/java/com/facebook/airlift/http/server/StatsRecordingHandler.java

```diff
@@ -23,21 +23,31 @@
 import java.util.concurrent.TimeUnit;
 
 import static java.lang.Math.max;
+import static java.util.Objects.requireNonNull;
 
 public class StatsRecordingHandler
         implements RequestLog
 {
     private final RequestStats stats;
+    private final boolean statsRecordEnabled;
```

Comment:
When this boolean is true, `RequestStats` is provided but not used. This is a bit weird.

Another way would just to make `stats` field `Optional`.

Reply:
I am thinking to just have 1 field (`Optional<RequestStats> stats`). So if `stats` is not present, is viewed as stats record is disabled

Reply:
Good point, updated.